### ISSUE TITLE
Proposing a new property for no-unused-vars to enforce Node-style error handling

### ISSUE THREAD

**Issue by tlhunter** [eslint/eslint#9112]
**Please describe what the rule should do:**
This rule is an extension of `no-unused-vars` as it relates to arguments. This would be similar to the existing existing **after-used** property, but not be as restrictive as the **all** property. The **all** property can be too restrictive, e.g. we only need to access a third parameter but not a second. The **after-used** property can be too permissive as we can ignore error parameters.

The existing **varsIgnorePattern** property is a blacklist of argument names and I do not believe can satisfy this proposal. However a whitelist property could satisfy this proposal.

**What category of rule is this? (place an "X" next to just one item)**
[ ] Enforces code style
[X] Warns about a potential error
[ ] Suggests an alternate way of doing something
[ ] Other (please specify:)

**Provide 2-3 code examples that this rule will warn about:**
```js
/*eslint no-unused-vars: ["error", { "args": "after-used-and-errors", "errors": ["e", "err", "error"] }]*/

// Eslint should complain as we're ignoring the error
asyncWork((err, data) => {
  data.xyz = true;
  callback(null, data);
});

// Eslint should not complain; we're handling the error, and making use of the final argument
asyncWork((err, ignorable, data) => {
  if (err) {
    return callback(err);
  }
  
  data.xyz = true;

  callback(null, data);
});

// Eslint should not complain; we're making use of the final argument
function somethingSync(ignorable, realData) { // e.g. existing args: after-used behavior
  return realData++;
}
```

**Why should this rule be included in ESLint (instead of a plugin)?**
The defacto Node.js standard for handling asynchronous errors is to call a callback function with the first argument being an error and latter arguments being data. Synchronous functions do not follow this standard. This rule would enforce that engineers need to handle the errors instead of ignoring them. This rule would be beneficial for all Node developers.

**Comment by tlhunter** [comment]
This rule already exists! http://eslint.org/docs/rules/handle-callback-err
